src/Symfony/Component/Intl/Locale.php

```diff
@@ -68,7 +68,14 @@ public static function getDefaultFallback(): ?string
     public static function getFallback(string $locale): ?string
     {
         if (\function_exists('locale_parse')) {
-            $localeSubTags = locale_parse($locale);
+            if (null === $localeSubTags = locale_parse($locale)) {
+                if (\defined('INTL_MAX_LOCALE_LEN')) {
+                    $message = sprintf('The length of $locale must be less than %s', \INTL_MAX_LOCALE_LEN);
+                }
+
+                throw new \InvalidArgumentException($message ?? '');
```

Comment:
Why throw and not return null instead? this would make more sense to me.

Reply:
This is @ro0NL recommendation for this issue.

Reply:
The description says: `The ICU locale code of the fallback locale, or null if no fallback exists`
`null` looks expected to me...

Reply:
we shouldnt hide errors IMHO

Reply:
why would this be an error? it's not, even for locale_parse

Reply:
id say  a valid locale can be parsed: https://3v4l.org/Q4RUV

otherwise i'd say the parent of a "too long" locale is still `root`

Reply:
Looking at the logic below, this might be the most expected:
`return 'root' !== self::$defaultFallback && self::$defaultFallback === $locale ? 'root' : null;`

Reply:
I'm doubling on this. @ro0NL OK for you?
@svperfecta can you please update?

Reply:
currently an error already bubbles up. The null check just polishes that.

But i think we can aim for consistency yes, accounting for `return self::$defaultFallback;` as well:

```php
if (null === $localeSubTags = locale_parse($locale)) {
    $localeSubTags = ['language' => $locale];
}
```

and let it go from there.